setup.py

```diff
@@ -31,6 +33,99 @@
         version = match.group(1)
 
 
+def _get_help_output(topic, lines=False):
+    TRANSLATE = {
+        'tar-content': 'tar_content',
+    }
+    if topic in TRANSLATE:
+        topic = TRANSLATE[topic]
+    return _run_snapcraft(['help', topic], lines)
+
+
+def _run_snapcraft(cmd, lines=False):
+    cmd = ['./bin/snapcraft']+cmd
+    command = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+    output = command.communicate()[0]
+
+    if sys.version_info.major == 3:
+        output = output.decode('utf-8')  # bytes to str
+    if lines:
+        return output.split('\n')[:-1]  # drop last empty line
+    return output
+
+
+class ReferenceCommand(Command):
+    DEFAULT_REFERENCE_FILE = 'docs/reference.md'
+    description = "build reference documentation from snapcraft's help command"
+    user_options = [
+        ('output-file=', 'O',
+         'Specify the output file for the reference, by '
+         'default: {}'.format(DEFAULT_REFERENCE_FILE))
+    ]
+    docs = []
+    topics = []
+    plugins = []
+
+    def initialize_options(self):
+        self.output_file = None
+
+    def finalize_options(self):
+        if not self.output_file:
+            self.output_file = os.path.abspath(self.DEFAULT_REFERENCE_FILE)
+
+    def run(self):
+        assert os.getcwd() == os.path.abspath(os.path.dirname(__file__)), \
+            'Must be run in snapcraft root.'
+        assert os.path.isdir(os.path.dirname(self.output_file)), \
+            'Directory {} does not exist.'.format(
+                os.path.dirname(self.output_file))
+        self.topics = _get_help_output('topics', lines=True)
+        self.plugins = _run_snapcraft(['list-plugins'], lines=True)
+        self.topics.extend(self.plugins)
+        for topic in self.topics:
+            text = _get_help_output(topic)
+            if not text.startswith('The plugin has no documentation'):
+                self.docs += [
+                    (topic, text)
+                ]
+        self._write_index_doc()
+
+    def _write_index_doc(self):
+        text = '''
```

Comment:
Use three """ instead, also, maybe call this text_template or something... @elopio will have ideas wrt to consistency on this one.